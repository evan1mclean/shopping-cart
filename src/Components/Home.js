import { Link } from "react-router-dom";
import shopping from "../images/shopping.jpg";

function Home() {
  return (
    <div
      className="grow flex lg:justify-between lg:items-center md:items-center w-full lg:px-24 px-10 py-10" 
    >
      <div className="lg:h-96 lg:w-1/2 flex flex-col lg:justify-between md:justify-between gap-7">
        <h1 className="lg:text-5xl md:text-5xl text-3xl font-bold lg:text-left text-center">
          Welcome To The Big Fake Online Storefront
        </h1>
        <p className="text-lg">
          Generic slogan for all your fake shopping needs... blah blah.. Lorem
          ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
          tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
          veniam, quis amet, consectetur adipiscing elit
        </p>
        <Link to={"/products"}>
          <div className="text-center lg:text-left">
            <button className="bg-sky-400 text-white text-lg py-2 px-4 rounded-3xl hover:scale-110 transition duration-100">
              Shop Now
            </button>
          </div>
        </Link>
      </div>
      {<div>
        <img src={shopping} alt="Online Shopping" className="lg:h-96 lg:w-auto rounded-lg h-0 w-0" />
      </div>}
    </div>
  );
}

export default Home;
