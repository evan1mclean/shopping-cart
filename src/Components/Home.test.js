import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Home from "./Home";
import { BrowserRouter } from "react-router-dom";

describe("Home", () => {
  test("Renders home page", () => {
    render(<Home />, {wrapper: BrowserRouter})
    expect(screen).toMatchSnapshot();
  })

  test("Renders home page header", () => {
    render(<Home />, {wrapper: BrowserRouter})
    expect(screen.getByTestId("home-header")).toBeInTheDocument();
  })

  test("Renders store description", () => {
    render(<Home />, {wrapper: BrowserRouter})
    expect(screen.getByTestId("store-description")).toBeInTheDocument();
  })

  test("Renders shop now button", () => {
    render(<Home />, {wrapper: BrowserRouter})
    expect(screen.getByTestId("shop-now-button")).toBeInTheDocument();
  })
})