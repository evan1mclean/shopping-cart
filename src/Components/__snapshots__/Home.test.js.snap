// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Renders home page 1`] = `
<div>
  <div
    class="grow flex lg:justify-between lg:items-center md:items-center w-full lg:px-24 px-10 py-10"
  >
    <div
      class="lg:h-96 lg:w-1/2 flex flex-col lg:justify-between md:justify-between gap-7"
    >
      <h1
        class="lg:text-5xl md:text-5xl text-3xl font-bold lg:text-left text-center"
        data-testid="home-header"
      >
        Welcome To The Big Fake Online Storefront
      </h1>
      <p
        class="text-lg"
        data-testid="store-description"
      >
        Generic slogan for all your fake shopping needs... blah blah.. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis amet, consectetur adipiscing elit
      </p>
      <div
        class="text-center lg:text-left"
      >
        <a
          href="/products"
        >
          <button
            class="bg-sky-400 text-white text-lg py-2 px-4 rounded-3xl hover:scale-110 transition duration-100"
            data-testid="shop-now-button"
          >
            Shop Now
          </button>
        </a>
      </div>
    </div>
    <div>
      <img
        alt="Online Shopping"
        class="lg:h-96 lg:w-auto rounded-lg h-0 w-0"
        src="shopping.jpg"
      />
    </div>
  </div>
</div>
`;
